packages/admin-app/src/components/ElectionSetupWizard/Steps/JointKeySelectStep.tsx

```diff
@@ -59,25 +60,29 @@ const JointKeySelectStep: React.FC<JointKeySelectStepProps> = ({ onNext, onChang
         }
     };
 
-    const ceremonyClient = ClientFactory.GetCeremonyClient();
+    const ceremonyClient = useCeremonyClient();
+    useEffect(() => {
+        const findKeyCeremonies = async () => {
+            await ceremonyClient.find(0, 100, { filter: {} }).then((response) => {
+                setKeyCeremonies(response.key_ceremonies);
+            });
+        };
 
-    const getKeyCeremonies = async () => {
-        await ceremonyClient.find(0, 100, { filter: {} }).then((response) => {
-            setKeyCeremonies(response.key_ceremonies);
-        });
-    };
+        findKeyCeremonies();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
@keithrfung as you can see I converted the client retrieval code into a hook, but the problem is hooks have to be at the root level and can't be called from inside other hooks.  So you have to pass it in to the array in the 2nd parameter of useEffect, but that produces infinite recursion.  The only solution I could find was this lint disable.  Got any ideas?

Reply:
This is probably the documentation you're looking for explaining the linting
https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies

Reply:
The ceremonyClient should not be changing. So by passing it in, it shouldn't be rerendering. If that is happening, there is something happening with the client itself. 

https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects